controllers/update_pods_test.go

```diff
@@ -246,5 +249,70 @@ var _ = Describe("update_pods", func() {
 				Expect(updates).To(HaveLen(1))
 			})
 		})
+
+		When("max unavailable pods is set with a percent value and there are process groups with pods in pending status", func() {
```

Comment:
Can we add a test case where we actually update some Pods even if some of them are unavailable?